### PR TITLE
Update with the new changes in core data retriever

### PR DESCRIPTION
**Pull request by apoorvaeternity**
Supersedes #52 

TODO:
- [ ] Test

### CODE REVIEW

**Comment by henrykironde** [comment]
@apoorvaeternity when you get time, could you merge the PR I sent https://github.com/apoorvaeternity/Retriever.jl/pull/1

File: test/test_retriever.jl [hunk 1]
```diff
@@ -79,6 +79,17 @@ function empty_files(path, ext)
     end
 end
 
+function dataset_name_upstream():
```
Comment (henrykironde) [inline test/test_retriever.jl]: I don't know how this made it's way here in my last commit ` :`

Comment (apoorvaeternity) [reply test/test_retriever.jl]: Looks like from python :smile: . I will fix it.

**Comment by henrykironde** [comment]
Thanks @apoorvaeternity for the great work on this PR.

**Pull request merged**